deployment.adoc

```diff
@@ -0,0 +1,114 @@
+purposes:
+
+Tendrl Master::
+* web server
+* tendrl api, proxied behind the web server
+* tendrl ui, served via the web server
+* etcd
+
+Storage Node::
+* storage system (ceph, gluster etc.)
+* tendrl node agent
+* tendrl bridge (node agent will install and configure the bridge as
+appropriate, depending upon the storage system)
+
+==== Post Deployment
+
+The following endpoints should be available for client consumption:
+
+frontend:: http://tendrl_master_host/
+api:: http://tendrl_master_host/api/1.0/
+
+
+== Component Installation
+
+=== etcd
+
+NOTE: All the commands are run as a regular user that has `sudo` privileges.
+
+IMPORTANT: The etcd connection details are required for deploying all the other
+components.
+
+* Install https://github.com/coreos/etcd/releases/tag/v2.3.7[etcd >= 2.3.x && <
+3.x].
++
+On CentOS 7, etcd is available in the `extras` repository.
+
+ $ sudo yum install etcd
+
+* By default, etcd will listed on the loopback interface. Edit the etcd
+configuration file `/etc/etcd/etcd.conf` to configure it to listen on the
+public interface.
+* Ensure that the firewall configuration allows inbound connections to etcd.
+* Enable and start the service
+
+ $ sudo systemctl enable etcd.service
+ $ sudo systemctl start etcd.service
+
+
+=== From Source on CentOS 7
+
+NOTE: The definition and flow yaml files are uploaded to etcd by the node
+agents upon starting up. The api application reads these yaml files and brings
+up the endpoints based on them.
+
+Node Agent::
+Install the node agent on every node of the storage cluster using the
+documentation at https://github.com/Tendrl/node_agent/blob/master/README.adoc.
+
+Tendrl API::
+The api installation instructions are in the
+https://github.com/Tendrl/tendrl-api/blob/master/README.adoc[api repository].
+Only follow the instructions from the sections listed below:
+  . https://github.com/Tendrl/tendrl-api/blob/master/README.adoc#system-setup[System Setup]
+  . https://github.com/Tendrl/tendrl-api/blob/master/README.adoc#install-tendrl-api[Install tendrl-api].
+    * Install only the production dependencies if you want only to run the api.
+    * Install everything if you want to also have the development setup as well.
+  . https://github.com/Tendrl/tendrl-api/blob/master/README.adoc#configuration[Configuration].
+  . https://github.com/Tendrl/tendrl-api/blob/master/README.adoc#running-on-port-80[Running on Port 80].
+
+Tendrl Frontend::
++
+NOTE: Currently, the frontend is served via apache using the same configuration
+file that ships with and is used by the api, as discussed in Step 2 under the
+https://github.com/Tendrl/tendrl-api/blob/master/README.adoc#running-on-port-80[instructions
+for running the api on port 80].
++
+Before deployment, the frontend needs to be compiled first. The compilation
+instructions are in the section `Step 2: Building from source` in the
+https://github.com/Tendrl/tendrl_frontend/blob/master/docs/deployment.adoc[frontend
```

Comment:
In the frontend deployment doc, I think the sections like Serve using nginx, Serve using httpd etc doesn't make much sense because it is kind of covered in - https://github.com/Tendrl/tendrl-api/blob/master/README.adoc#running-on-port-80  - which is good enough.